std/string.d

```diff
@@ -5798,36 +5798,57 @@ unittest
  * function, or any of the conversion functions.
  *
  * Params:
- *     s = the string to check
+ *     s = the string or random access range to check
  *     bAllowSep = accept separator characters or not
  *
  * Returns:
  *     $(D bool)
  */
-bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S)
+bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S ||
+    (isRandomAccessRange!S &&
+    hasSlicing!S &&
+    isSomeChar!(ElementEncodingType!S) &&
```

Comment:
`ElementEncodingType!S` is the same as `ElementType!S` when `isSomeString!S` is false, isn't it? I suggest to go with `ElementType!S` then.